asyncio_mqtt/client.py

```diff
@@ -218,6 +227,18 @@ async def force_disconnect(self) -> None:
         if not self._disconnected.done():
             self._disconnected.set_result(None)
 
+    def _outgoing_call(self, func: Callable) -> Callable[..., Awaitable]:
```

Comment:
Well done with this decorator. :+1: 

One nitpick: The outgoing call may return something different from `None` (e.g., `subscribe` returns `int`). Let's preserve the return type via a `TypeVar`. Similar to what [they do in this example from the standard library](https://docs.python.org/3/library/typing.html#typing.ParamSpec). Note that we don't have `ParamSpec` yet (It's in 3.10) but we can still preserve the return type.

Reply:
What about https://github.com/sbtinstruments/asyncio-mqtt/pull/101/commits/7274933734c3b1568e662e3b1934e8171295d86e (with an explicit explanation of what types are possible)?

Reply:
Good question. :) I prefer `TypeVar` over `Union` due to type scope. See https://stackoverflow.com/a/58903907/554283 for reference. Let me know if it makes sense or not.

Reply:
https://github.com/sbtinstruments/asyncio-mqtt/pull/101/commits/b6e5b680f88f38742c057fe9cb3f13e726ffceb2?

Reply:
Something like this (please test it first):

```python
T = TypeVar("T")

def _outgoing_call(self, func: Callable[..., Awaitable[T]]) -> Callable[..., Awaitable[T]]:
    @wraps(func)
    async def _decorated(*args: Any, **kwargs: Any) -> T:
        ....
    return _decorated
```

Reply:
This shows how badly I understand TypeVars... 😆 `mypy` doesn't complain about this, so I'm assuming it's good.

Reply:
Don't worry about it. It is always a pain to add typing to a decorator function. :))